playground/infrastructure/helper.py

```diff
@@ -57,10 +62,71 @@ def find_examples(work_dir: str) -> List[Example]:
     Returns:
         List of Examples.
     """
-    examples = [Example()]
+    examples = []
+    for root, _, files in os.walk(work_dir):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            if match_pattern(filepath):
+                name = get_name(filename)
+                sdk = get_sdk(filename)
+                with open(filepath) as parsed_file:
+                    content = parsed_file.read()
+
+                example = Example(name, sdk, filepath, content)
+                examples.append(example)
     return examples
 
 
+def match_pattern(filepath: str) -> bool:
+    """Check file to matching
+
+    Check that file has the correct extension and contains the beam-playground tag.
+
+    Args:
+        filepath: path to the file.
+
+    Returns:
+        True if file matched. False if not
+    """
+    extension = filepath.split(os.extsep)[-1]
+    if extension in SUPPORTED_SDK.keys():
+        with open(filepath) as parsed_file:
+            content = parsed_file.read()
+        return re.search(PATTERN, content) is not None
+
+
+def get_name(filename) -> str:
+    """ Return name of the example by his filepath.
+
+    Get name of the example by his filename.
+
+    Args:
+        filename: filename of the beam example file.
+
+    Returns:
+        example's name.
+    """
+    return filename.split(os.extsep)[0]
+
+
+def get_sdk(filename) -> Sdk:
+    """ Return SDK of example by his filename.
+
+    Get extension of the example's file and returns associated SDK.
+
+    Args:
+        filename: filename of the beam example.
+
+    Returns:
+        Sdk according to file extension.
+    """
+    extension = filename.split(os.extsep)[-1]
+    if SUPPORTED_SDK.get(extension) is not None:
```

Comment:
```suggestion
    if extension in SUPPORTED_SDK:
```
This construction is more readable.

Reply:
Changed.